redis-utils/src/main/java/org/ballerinalang/redis/actions/AbstractRedisAction.java

```diff
@@ -410,14 +408,14 @@ static long bitCount(String key, RedisDataSource<String, String> redisDataSource
         }
     }
 
-    static <K> BMap<K, BString> bLPop(long timeout, RedisDataSource<K, String> redisDataSource, K... keys) {
+    static <K> BMap<K, String> bLPop(long timeout, RedisDataSource<K, String> redisDataSource, K... keys) {
         RedisListCommands<K, String> redisCommands = null;
         try {
             redisCommands = (RedisListCommands<K, String>) getRedisCommands(redisDataSource);
             KeyValue<K, String> result = redisCommands.blpop(timeout, keys);
             if (result != null) {
-                BMap<K, BString> bMap = new MapValueImpl<>();
-                bMap.put(result.getKey(), StringUtils.fromString(result.getValue()));
+                BMap<K, String> bMap = new MapValueImpl<>();
```

Comment:
Shouldn't this be `MapValue<K, String>` ? Please check the other places as well.

Reply:
Updated, including other places